### DISCUSSION TITLE
newlines flapping in ".po" headers

### DISCUSSION THREAD

**Discussion by cfm** [WeblateOrg/weblate#7791] [Q&A] [answered]
We show newlines flapping in `.po` headers between two different stages in our Weblate workflow:

1. On translation change, Weblate flattens newlines:  https://github.com/freedomofpress/securedrop-client/commit/1b9793bb2498ff99f4e965296ea88e2a2f682782
2. On source-string change, the `weblate.gettext.msgmerge` add-on expands newlines:  https://github.com/freedomofpress/securedrop-client/commit/4fec345bfe211c3a25e51d273f5485ea2d5b2fc2

(2) is the desired behavior.  Does this seem like a misconfiguration in Weblate or our add-ons; a Weblate bug; or a ["newlines round-trip"‒type bug](https://github.com/translate/translate/search?q=newlines+round+trip&type=issues) in the Translate Toolkit?

**Comment by nijel** [comment, marked as answer]
Do you have [Customize Gettext add-on](https://docs.weblate.org/en/latest/admin/addons.html#customize-gettext-output) installed? What configuration does it use?

**Reply by cfm** [reply]
Thanks, @nijel; this was it.  I had to understand <https://github.com/WeblateOrg/weblate/blob/b86f794dbd620c419ee93f007629ebc1826745a0/weblate/addons/forms.py#L116-L123> a bit better to understand what our settings here _should_ be, versus what they've historically been.

I'll follow up with a small pull request to clarify the `GettextCustomizeForm.width.choices` labels a bit further.  :-)

**Reply by nijel** [reply]
Still, the msgmerge add-on should honor the settings from the customize add-on, so that doesn't seem to work as expected...
